techLogger.js

```diff
@@ -72,47 +74,29 @@ function _makeLogger(prefix) {
             return updated;
         },
 
-        debug: function() {
-            _log("debug", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
-        },
+        debug: (...args) => _log("debug", _prefixLog(args, logPrefix)),
 
-        info: function() {
-            _log("info", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
-        },
+        info: (...args) => _log("info", _prefixLog(args, logPrefix)),
 
-        notice: function() {
-            _log("notice", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
-        },
+        notice: (...args) => _log("notice", _prefixLog(args, logPrefix)),
 
-        warn: function() {
-            _log("warning", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
-        },
+        warn: (...args) => _log("warning", _prefixLog(args, logPrefix)),
 
-        error: function() {
-            _log("error", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
-        },
+        error: (...args) => _log("error", _prefixLog(args, logPrefix)),
 
-        crit: function() {
-            _log("crit", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
-        },
+        crit: (...args) => _log("crit", _prefixLog(args, logPrefix)),
 
-        alert: function() {
-            _log("alert", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
-        },
+        alert: (...args) => _log("alert", _prefixLog(args, logPrefix)),
 
-        emerg: function() {
-            _log("emerg", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
+        emerg(...args) {
+            return _log("emerg", _prefixLog(args, logPrefix));
         },
 
-        createExpressLoggerStream: function(level) {
-            return {
-                write: function(message /*, encoding */ ) {
-                    _log(level, message);
-                }
-            };
+        createExpressLoggerStream(level) {
+            return { write: (message /*, encoding */ ) => _log(level, message) };
         },
 
-        splash: function(app, configuration) {
+        splash(app, configuration) {
             splash(this, app, configuration);
         }
```

Comment:
@KINFOO Thanks for your feedback, going further with your approach we can have:

```js

const LOG_LEVELS = ["emerg", "alert", "crit", "error", "notice", "warn", "info", "debug"];

function _makeLogger(prefix) {
    var logger = {
       //....
       splash(app, configuration) {
            splash(this, app, configuration);
        }
    };
    LOG_LEVELS.forEach(level => {
        logger[level] = (...args) => _log(level, _prefixLog(args, logPrefix));
    });

    // And even shorter (still readable?)
    LOG_LEVELS.forEach(level => logger[level] = (...args) => _log(level, _prefixLog(args, logPrefix)));

    return logger;
}
```

@sgaestel it could have been nice, but it does not work

```
/home/sebz/dev/swir/av/node-tech-logger/lib/splash.js:11
    if (configuration.logging.keysToHide) {
                     ^
TypeError: Cannot read property 'logging' of undefined
    at Function.module.exports [as splash] (/home/sebz/dev/swir/av/node-tech-logger/lib/splash.js:11:22)
    at Object.<anonymous> (/home/sebz/dev/swir/av/node-tech-logger/test/testSplash.js:9:8)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.runMain (module.js:605:10)
    at run (bootstrap_node.js:420:7)
    at startup (bootstrap_node.js:139:9)
npm ERR! Test failed.  See above for more details.
```

But thanks to your remark, I'll rework this function to have something less confusing :sweat_smile: